src/Avalonia.Base/Platform/PlatformBehavior.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Avalonia.Platform
+{
+    internal class PlatformBehaviorInhibitionDisposer : IDisposable
+    {
+        readonly Action _disposeAction;
+
+        public PlatformBehaviorInhibitionDisposer(Action disposeAction)
+            => _disposeAction = disposeAction;
+
+        public void Dispose()
+            => _disposeAction();
+    }
+
+    public enum PlatformBehaviorInhibitionType
+    {
+        AppSleep
+    }
+
+    public interface IPlatformBehaviorInhibition
```

Comment:
Can you change the way how this interface is registered and used.
In latest preview we have introduced a new pattern how we want to use platform specific features instead of global service locator. As well as avoid static APIs in general.

ITopLevelImpl has now TryGetFeature method.
Then you can add 
```csharp
Task<IDisposable> RequestPlatformInhibition(PlatformInhibitionType type, string reason);
```
to the TopLevel, which internally will attempt to resolve platform implementation using `PlatformImpl.TryGetFeature<IPlatformBehaviorInhibition>()`.

Reply:
Also, new API needs XML documentation.